pkg/resources/kafka/pod.go

```diff
@@ -90,6 +90,7 @@ func (r *Reconciler) pod(id int32, brokerConfig *v1beta1.BrokerConfig, pvcs []co
 			util.MergeLabels(
 				kafkautils.LabelsForKafka(r.KafkaCluster.Name),
 				map[string]string{"brokerId": fmt.Sprintf("%d", id)},
+				brokerConfig.BrokerLabels,
```

Comment:
We need to ensure that users can not override reserved labels that `koperator` relies on such as `brokerId` and the labels returned by `kafkautils.LabelsForKafka`. If those labels are present in `brokerConfig.BrokerLabels` then those needs to be ignored (this needs to be stated in the description of the `brokerLabels` field). This can simply achieved by passing in `brokerLabels` as the first argument to `util.MergeLabels()` and also adding a unit test that verifies that users can't override reserved labels.


For consistency there could be `brokerConfig.GetBrokerLabels()` method similarly to `brokerConfig.GetBrokerAnnotations()`